server/src/com/thoughtworks/go/server/controller/JobController.java

```diff
@@ -154,10 +141,12 @@ public ModelAndView handleRequest(@RequestParam(value = "pipelineName")String pi
                                       HttpServletResponse response) {
         Json json;
         try {
-            JobInstance instance = jobInstanceService.buildByIdWithTransitions(jobId);
-            JobStatusJsonPresentationModel presenter = new JobStatusJsonPresentationModel(instance,
-                    goConfigService.agentByUuid(instance.getAgentUuid()),
-                    stageService.getBuildDuration(pipelineName, stageName, instance));
+            JobInstance requestedInstance = jobInstanceService.buildByIdWithTransitions(jobId);
+            JobInstance mostRecentJobInstance = jobDetailService.findMostRecentBuild(requestedInstance.getIdentifier());
```

Comment:
I think someone closer should review this. :) I think it looks good. The only thing I'd think about is the impact on performance. This is called from the jobs detail page, right? I see that after #1130 was merged, the jobStatus.json call never stops, even after a build has gone green. It might be useful to fix that too, so that this call does not turn out to be a bottleneck.

/cc @ketan, what do you think?